### PR TITLE
formula_installer: set `HOMEBREW_RELOCATE_RPATHS`

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally? (In progress...)

-----

This enables `RPATH` relocation upon pouring bottles.

Part of #11329.

### CODE REVIEW

File: Library/Homebrew/formula_installer.rb [hunk 1]
```diff
@@ -337,6 +337,10 @@ def build_bottle_postinstall
 
   sig { void }
   def install
+    # TODO: Remove when bottles are built with RPATH relocation enabled
+    # https://github.com/Homebrew/brew/issues/11329
+    ENV["HOMEBREW_RELOCATE_RPATHS"] = "1"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_installer.rb:342]: Should this be done in `pour` perhaps instead (ideally as close as possible to the relevant call site)? Also, need to make sure it gets unset after it's no longer needed. `with_env` may be able to help with that.

Comment (carlocab) [reply Library/Homebrew/formula_installer.rb]: Good suggestion. Done.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by carlocab** [comment]
Failures are cask-related. Will try to re-run tests.

**Comment by carlocab** [comment]
Rebasing for fixes from #11343.

**Pull request merged**